### PR TITLE
Remove long usage in Compiler 

### LINKED ISSUE #2274: Old script compiler: clean up "long" variables
The old compiler works with "long" type alot, its code probably was originally written having 16-bit systems in mind.

"Long" type is non-portable, as it has different sizes on 64-bit Windows and Unix platforms. It's hard to tell whether this is actually causing any problems, but it's better to be safe imo.

This could be related to #2088, but only if Editor's native parts could be built on other platforms/compilers; so perhaps not so much.
On another hand, compiler may be built as a standalone program now.

### PR DESCRIPTION
**Pull request by ericoporto**
fix #2274

I tried to keep changes at minimum.

### CODE REVIEW

**Comment by ericoporto** [comment]
Did an additional commit just to kick the CI and see how it goes, but I think that the packfile struct was from the time the system was DOS and 16-bit so I instead switched things there from long to int.

File: Compiler/fmem.h [hunk 1]
```diff
@@ -19,16 +19,16 @@
...
 {
   char *data;
-  long len;                     // length of data in array
-  long size;                    // size of data array allocated
-  long magic;
-  long pos;                     // current seeked position
+  size_t len;                  // length of data in array
+  size_t size;                 // size of data array allocated
+  int magic;
+  int pos;                     // current seeked position
```
Comment (ivan-mogilko) [inline Compiler/fmem.h:25]: pos also should be size_t, as it should be comparable to size and len fields.

File: libsrc/allegro/include/allegro/file.h [hunk 2]
```diff
@@ -70,9 +70,9 @@ struct PACKFILE_VTABLE
    AL_METHOD(int, pf_fclose, (void *userdata));
    AL_METHOD(int, pf_getc, (void *userdata));
    AL_METHOD(int, pf_ungetc, (int c, void *userdata));
-   AL_METHOD(long, pf_fread, (void *p, long n, void *userdata));
+   AL_METHOD(int, pf_fread, (void *p, int n, void *userdata));
    AL_METHOD(int, pf_putc, (int c, void *userdata));
-   AL_METHOD(long, pf_fwrite, (AL_CONST void *p, long n, void *userdata));
+   AL_METHOD(int, pf_fwrite, (AL_CONST void *p, int n, void *userdata));
```
Comment (ivan-mogilko) [inline libsrc/allegro/include/allegro/file.h:75]: It's better to make these size_t, for compliance with classic stream reading and writing.

Comment (ericoporto) [reply libsrc/allegro/include/allegro/file.h]: Switched to size_t!

File: Compiler/fmem.h (hunk 1 above)
Comment (ericoporto) [reply Compiler/fmem.h]: Done

**Comment by ivan-mogilko** [comment]
Sorry, I made a mistake about allegro packfile!

I noticed now that it returns -1 or EOF (which is also -1) on error. This means that the return value is supposed to be signed.
Could you please revert this last commit and leave them as int32?
Otherwise we'd need to also fix return values, and scan all uses of these functions in libraries to see how the return values are checked.

**Comment by ericoporto** [comment]
That is fine, I will revert.

**Comment by ivan-mogilko** [comment]
>  I think the idea would be to return the maximum possible value of size_t?

Standard read and write return strictly number of processed bytes, so in case of EOF or error they should return 0 instead.

**Comment by ericoporto** [comment]
I decide to simply drop that commit (https://github.com/adventuregamestudio/ags/commit/aa77a424cc1886544ec9f9ccc8e755a13dcdafa3) for the time being, as I think it's not relevant now and the long in these files are a good reminder to update that at some point.

**Pull request merged**